### PR TITLE
ComboBox reset current on model changed

### LINKED ISSUE #3628: [ComboBox] setting index to -1 does not clear ComboBox text
Changing a ComboBox's index changes the text shown in the widget.
In case current-index is set to -1...
... expected behavior: current-text is reset.
... actual behavior: current-text remains unchanged

Example to reproduce issue:
https://slintpad.com/?snippet=import+%7B+Button%2C+VerticalBox+%2C+ComboBox%2C+SpinBox%7D+from+%22std-widgets.slint%22%3B%0Aexport+component+Demo+%7B%0A++++VerticalBox+%7B%0A++++++++Text+%7B%0A++++++++++++text%3A+%22SpinBox+value+%27-1%27+does+not+reset+ComboBox+text.%22%3B%0A++++++++++++font-size%3A+24px%3B%0A++++++++++++horizontal-alignment%3A+center%3B%0A++++++++%7D%0A++++++++combo-box+%3A%3D+ComboBox+%7B%0A++++++++++++model%3A+%5B%22Zero%22%2C%22One%22%2C%22Two%22%5D%3B%0A++++++++%7D%0A++++++++SpinBox+%7B%0A++++++++++++minimum%3A+-1%3B%0A++++++++++++maximum%3A+combo-box.model.length+-+1%3B%0A++++++++++++edited+%3D%3E+%7B%0A++++++++++++++++combo-box.current-index+%3D+self.value%3B%0A++++++++++++%7D%0A++++++++%7D%0A++++%7D%0A%7D%0A

### LINKED ISSUE #5214: Updating the `model` of a `ComboBox` does not change the currently displayed text
Changing the `model`, i.e. the list of options, in a `ComboBox` does not also change affect the `current-index` or the `current-value`. [Here](https://slintpad.com/?snippet=import+%7B+Button%2C+VerticalBox%2C+ComboBox+%7D+from+%22std-widgets.slint%22%3B%0Aexport+component+Demo+%7B%0A++++VerticalBox+%7B%0A++++++++combo+%3A%3D+ComboBox+%7B%0A++++++++++++model%3A+%5B%22Ex+1%22%2C+%22Ex+2%22%2C+%22Ex+3%22%2C+%22Ex+4%22%5D%3B%0A++++++++%7D%0A++++++++Button+%7B%0A++++++++++++text%3A+%22Change%22%3B%0A++++++++++++clicked+%3D%3E+%7B%0A++++++++++++++++combo.model+%3D+%5B%22Test+1%22%2C+%22Test+2%22%2C+%22Test+3%22%5D%3B%0A++++++++++++%7D%0A++++++++%7D%0A++++%7D%0A%7D%0A&style=native) is an demonstration of the issue in SlintPad.

In particular, the text in the box still shows the selected value from the previous model. Clicking the box and selecting one of the new items will fix the issue. Closing the selection box without making a new selection results in the text from the old model still being displayed.

It is possible that this is the intended behavior, but it seems to me that changing the model should also do the following:
1. Set the `current-value` to the text of first item in the new model so it at least is consistent with the new model.
2. Set the `current-index` either to 0 (the first new item) or  -1 (no item selected).

For what it's worth, I am using Rust in both Windows and Linux, and this issue occurs on all platforms (including SlintPad). For now I have a workaround that performs the above steps, but it is a bit hacky and manual, but will made easier and more automatic once [this feature](https://github.com/slint-ui/slint/issues/112) is fully implemented and released.

### PR DESCRIPTION
**Pull request by FloVanGH**
Fixes #5214
Fixes #3628

### CODE REVIEW

**Review by ogoffart — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -31,6 +31,8 @@ All notable changes to this project are documented in this file.
  - Fixed compiler panic when a callback aliases itself. (#4938)
  - Fixed compiler panic when an init from a repeater is inlined into the parent component (#5146)
  - Added `clear-focus()` function to focusable elements, to allow for programmatic focus clearing.
+ - Fixed updating model of ComboBox does not change current-value
+ - Fixed set current-index of ComboBox to -1 does not reset current-value
```
Comment (ogoffart) [inline CHANGELOG.md:35]: These actually belong in the Widget category.

Be carefull as well that if this is not merged before the release, it needs to be changed to a different version manually.

Comment (FloVanGH) [reply CHANGELOG.md:35]: ok

**Comment by tronical** [comment]
I tried to debug this a little. It’s the inlining that breaks with change handlers. SLINT_INLINING=true reproduces the failure also for builds (gallery for example). It looks like the names references inside the change handlers aren’t adjusted correctly. From the code it looks like that should happen when processing the fix ups for the mapping, but something goes wrong.

**Comment by tronical** [comment]
Minimal reproducer:

```slint
component Test {
    in-out property <int> index: 0;
    changed index => {
        debug(root.index);
    }
}

export component Testcase  {
    in-out property index <=> t.index;
    t := Test {}
}
```

**Pull request merged**